### PR TITLE
Add types to missing files in `lib/route`.

### PR DESCRIPTION
**Pull request by sgomes**
Added types to `legacy-routes` and `path`.

#### Changes proposed in this Pull Request

* Added types to `legacy-routes` and `path`.

#### Testing instructions

* Ensure the unit tests continue running correctly.

### CODE REVIEW

File: client/lib/route/path.ts [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
  */
-import { includes } from 'lodash';
+import { URL as URLString, SiteSlug, SiteId } from 'types';
 
 /**
  * Internal Dependencies
  */
 import { trailingslashit, untrailingslashit } from './index';
 
+type SiteFragment = SiteSlug | number | false;
```
Comment (sirreal) [inline client/lib/route/path.ts]: I think we're actually better off not defining this and using `SiteId`, `SiteSlug` and `false` where they're expected.
```suggestion
```

File: client/lib/route/path.ts [hunk 2]
```diff
@@ -21,7 +22,7 @@ const statsLocationsByTab = {
 	googleMyBusiness: '/google-my-business/stats/',
 };
 
-export function getSiteFragment( path ) {
+export function getSiteFragment( path: URLString ): SiteFragment {
```
Comment (sirreal) [inline client/lib/route/path.ts]:
```suggestion
export function getSiteFragment( path: URLString ): SiteSlug | SiteId | false {
```

File: client/lib/route/path.ts [hunk 3]
```diff
@@ -85,7 +86,7 @@ export function addSiteFragment( path, site ) {
 	return pieces.join( '/' );
 }
 
-export function sectionify( path, siteFragment ) {
+export function sectionify( path: URLString, siteFragment?: SiteFragment ): URLString {
```
Comment (sirreal) [inline client/lib/route/path.ts]: Here, `false` would be a strange thing to pass in. There's a truthy check that's presumably expected to handle the `undefined` case, but passing in a `false` `SiteFragment` would also hit that case. Let's change this to:

```suggestion
export function sectionify( path: URLString, siteFragment?: SiteSlug | SiteId ): URLString {
```

File: client/lib/route/path.ts [hunk 4]
```diff
@@ -102,17 +103,20 @@ export function sectionify( path, siteFragment ) {
 	return untrailingslashit( basePath );
 }
 
-export function getStatsDefaultSitePage( slug ) {
+export function getStatsDefaultSitePage( slug?: SiteId | SiteSlug ): URLString {
```
Comment (sirreal) [inline client/lib/route/path.ts]: `slug` is a funny name when you put the type next to it, but that's fine 🙂

File: client/lib/route/path.ts [hunk 5]
```diff
@@ -102,17 +103,20 @@ export function sectionify( path, siteFragment ) {
...
+		return path + String( slug );
 	}
 
 	return untrailingslashit( path );
 }
 
-export function getStatsPathForTab( tab, siteIdOrSlug ) {
+export function getStatsPathForTab(
+	tab: keyof typeof statsLocationsByTab,
+	siteIdOrSlug: SiteId | SiteSlug
```
Comment (sirreal) [inline client/lib/route/path.ts]: 👍 This is nice

**Review by sirreal — APPROVED** [review]
Thanks for adding types 🙌 

I left some suggestions I think would improve things, but happy to see this ship when you think it's ready.

File: client/lib/route/path.ts (hunk 4 above)
Comment (sgomes) [reply client/lib/route/path.ts]: Renamed to `siteFragment`, to match the other instances 👍

**Comment by sgomes** [comment]
Thanks for the review, @sirreal! I applied the changes you suggested.

**Pull request merged**